tests/test_stability_funs.py

```diff
@@ -349,3 +351,402 @@ def test_magwell_print(capsys):
+def test_ballooning_compare_with_COBRAVMEC():
+    """Compare marginal stability points from DESC ballooning solve with COBRAVMEC.
+
+    COBRAVMEC uses some higher order techniques to refine the growth rate value,
+    so we don't expect the actual values to agree, but the sign should be the same.
+    so instead of comparing growth rates, we compare the radial point where it
+    becomes unstable. Recall that it should be stable on axis (zero pressure gradient)
+    but can become unstable elsewhere.
+    """
+
+    def find_root_simple(x, y):
+        sign_changes = np.where(np.diff(np.sign(y)))[0]
+
+        if len(sign_changes) == 0:
+            return None  # No zero crossing found
+
+        # Get the indices where y changes sign
+        i = sign_changes[0]
+
+        # Linear interpolation
+        x0, x1 = x[i], x[i + 1]
+        y0, y1 = y[i], y[i + 1]
+
+        # Calculate the zero crossing
+        x_zero = x0 + (0 - y0) * (x1 - x0) / (y1 - y0)
+
+        return x_zero
+
+    A = np.loadtxt("./tests/inputs/cobra_grate.HELIOTRON_L24_M16_N12")
+
+    ns1 = int(A[0, 2])
+    nangles = int(np.shape(A)[0] / (ns1 + 1))
+
+    B = np.zeros((ns1,))
+    for i in range(nangles):
+        if i == 0:
+            B = A[i + 1 : (i + 1) * ns1 + 1, 2]
+        else:
+            B = np.vstack((B, A[i * ns1 + i + 1 : (i + 1) * ns1 + i + 1, 2]))
+
+    gamma1 = np.amax(B, axis=0)
+
+    s1 = np.linspace(0, 1, ns1)
+    s1 = s1 + np.diff(s1)[0]
+
+    # COBRAVMEC calculated everything in s(=rho^2),
+    # DESC calculates in rho(=sqrt(s))
+    rho1 = np.sqrt(s1)
+
+    root_COBRAVMEC = find_root_simple(rho1, gamma1)
+
+    eq = desc.examples.get("HELIOTRON")
+
+    # Flux surfaces on which to evaluate ballooning stability
+    surfaces = [0.98, 0.985, 0.99, 0.995, 1.0]
+
+    grid = LinearGrid(rho=jnp.array(surfaces), NFP=eq.NFP)
+
+    Nalpha = 8  # Number of field lines
+
+    assert Nalpha == int(8), "Nalpha in the compute function hard-coded to 8!"
+
+    # Field lines on which to evaluate ballooning stability
+    alpha = jnp.linspace(0, np.pi, Nalpha + 1)[:Nalpha]
+
+    # Number of toroidal transits of the field line
+    ntor = 3
+
+    # Number of point along a field line in ballooning space
+    N0 = 2 * ntor * eq.M_grid * eq.N_grid + 1
+
+    # range of the ballooning coordinate zeta
+    zeta = np.linspace(-jnp.pi * ntor, jnp.pi * ntor, N0)
+
+    lam2_array = np.zeros(
+        len(surfaces),
+    )
+    for i in range(len(surfaces)):
+        rho = surfaces[i]
+        grid = Grid.create_meshgrid(
+            [rho, alpha, zeta],
+            coordinates="raz",
+            period=(np.inf, 2 * np.pi, np.inf),
+        )
+
+        data_keys = ["ideal ball lambda"]
+        data = eq.compute(data_keys, grid=grid)
+
+        lam2_array[i] = np.max(data["ideal ball lambda"])
+
+    root_DESC = find_root_simple(np.array(surfaces), lam2_array)
+    np.testing.assert_allclose(root_COBRAVMEC, root_DESC, atol=5e-4, rtol=1e-8)
```

Comment:
Same comment as above, use `atol=0` and increase `rtol` if necessary.

Reply:
This is  different test from above as we are comparing two very different solvers. Unfortunately, default atol will not work.

Reply:
What are the raw values for `root_COBRAVMEC` and `root_DESC`? I want to make sure they aren't less than the the absolute tolerance you are using. You should always be able to make it work with `atol=0` if `rtol` is large enough, unless the correct value is very close to 0.

Reply:
Around ~0.98. 

> You should always be able to make it work with atol=0 if rtol is large enough, unless the correct value is very close to 0.

Nah, it depends on the function. If you are trying to match a set of point corresponding to a functions a and b that cross zero then you can't match with any rtol if where the  function b is close to 0.

I have assumed 
abs(a-b) <= atol + rtol * abs(b)

Reply:
Try making `atol=0` and increase `rtol` to a few percent or whatever it needs to be

Reply:
Done.